tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -8720,6 +8721,9 @@ bool HLSLExternalSource::CanConvert(
           Second = ICK_HLSL_Derived_To_Base;
           goto lSuccess;
         }
+        // There is no way to cast to anonymous structures.  So we allow legacy
+        // HLSL implicit casts to matching anonymous structure types.
+        TargetUnnamed = !targetRD->hasNameForLinkage();
```

Comment:
Minor, but I find boolean variables with negations in them confusing. `TargetNamed` and flip the logic?

Reply:
Sure, I think originally had thought of it as "is anonymous" but it's based on this name for linkage property, so "Unnamed".  Alternatively, it could be "TargetAnon" Instead and don't flip the logic?

Reply:
Actually, I kind of like `TargetIsAnonymous` for clarity.